### PR TITLE
add Packet, BasePack and implemetnations (NetorkPacketMove)

### PR DESCRIPTION
**Pull request by ghost**
Requirements:
- add only packet + implementations + tests for serialization it in the future

### CODE REVIEW

File: common/src/main/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketMove.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import com.blackship.battlesheep.common.communication.packet.BasePacket;
+import com.blackship.battlesheep.common.communication.packet.Packet;
+import com.blackship.battlesheep.common.communication.packet.PacketMove;
+import com.blackship.battlesheep.common.communication.packet.enums.PacketType;
+
+import java.time.LocalTime;
+import java.util.HashSet;
+import java.util.Set;
+
+public class NetworkPacketMove extends BasePacket implements PacketMove {
```
Comment (mhbratek) [inline common/src/main/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketMove.java:12]: Other classes have JavaDoc, when you creating new file.

File: common/src/main/java/com/blackship/battlesheep/common/communication/network/packet/PacketFactory.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.blackship.battlesheep.common.communication.network.packet;
+
+import com.blackship.battlesheep.common.communication.packet.enums.PacketType;
+
+public class PacketFactory {
+
+    public static NetworkPacketMove createMove() {
+        return new NetworkPacketMove(PacketType.MOVE);
+    }
+
```
Comment (mhbratek) [inline common/src/main/java/com/blackship/battlesheep/common/communication/network/packet/PacketFactory.java:10]: Empty line

File: common/src/main/java/com/blackship/battlesheep/common/communication/packet/PacketMove.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.blackship.battlesheep.common.communication.packet;
+
+import java.util.Set;
+
+public interface PacketMove {
```
Comment (piotr921) [inline common/src/main/java/com/blackship/battlesheep/common/communication/packet/PacketMove.java:5]: Add comments

File: common/src/main/java/com/blackship/battlesheep/common/communication/packet/Packet.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.blackship.battlesheep.common.communication.packet;
+
+import com.blackship.battlesheep.common.communication.packet.enums.PacketType;
+
+import java.time.LocalTime;
+
+public interface Packet {
```
Comment (mhbratek) [inline common/src/main/java/com/blackship/battlesheep/common/communication/packet/Packet.java:7]: Add JavaDoc.

File: common/src/main/java/com/blackship/battlesheep/common/communication/packet/PacketMove.java (hunk 3 above)
Comment (mhbratek) [inline common/src/main/java/com/blackship/battlesheep/common/communication/packet/PacketMove.java:5]: Add JavaDoc.

File: common/src/main/java/com/blackship/battlesheep/common/communication/network/packet/PacketFactory.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package com.blackship.battlesheep.common.communication.network.packet;
+
+import com.blackship.battlesheep.common.communication.packet.enums.PacketType;
+
+public class PacketFactory {
```
Comment (piotr921) [inline common/src/main/java/com/blackship/battlesheep/common/communication/network/packet/PacketFactory.java:5]: Design patterns 👍

File: common/src/test/java/com/blackship/battlesheep/common/communication/packet/PacketTest.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        Packet givenPacket = PacketFactory.createMove();
+
+        //when
+        givenPacket.setCreationTime(expectedTime);
+        LocalTime givenTime = givenPacket.getCreationTime();
+
+        //then
+        assertEquals(givenTime, expectedTime);
+    }
+
```
Comment (mhbratek) [inline common/src/test/java/com/blackship/battlesheep/common/communication/packet/PacketTest.java:27]: Empty line.

File: common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketMoveTest.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+public class NetworkPacketMoveTest {
+
+    @DataProvider(name = "packetTypes")
+    public static Object[][] packetTypes() {
+        return new Object[][] { { PacketType.MOVE }, { PacketType.UPDATE }, { PacketType.ERROR } };
+    }
+
+    @Test(dataProvider = "packetTypes")
+    public void getPacketTypeFromPacketAndExpectGivenTypeTest(PacketType expectedPacketType) {
```
Comment (mhbratek) [inline common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketMoveTest.java:21]: Nice idea :+1:  Maybe we add it to our code.

**Review by piotr921 — APPROVED** [review]

File: common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketMoveTest.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+        //when
+        PacketType givenPacketType = new NetworkPacketMove(expectedPacketType).getPacketType();
+
+        //then
+        assertEquals(expectedPacketType, givenPacketType);
+    }
+
+    @Test
+    public void addPositionInPacketMoveAndExpectPutPositionInSetInPacketMoveTest() {
```
Comment (mhbratek) [inline common/src/test/java/com/blackship/battlesheep/common/communication/network/packet/NetworkPacketMoveTest.java:33]: Think about change naming convention
for example:  ```shouldAddGivenPositionInPacketMove```

Useful words: Given When Should

**Review by mhbratek — APPROVED** [review]
Add JavaDocs.
More info in commits messages, because now they say me nothing. 
Well done I your's code.

**Pull request merged**